flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/GraphCsvReader.java

```diff
@@ -0,0 +1,388 @@
+
+	public GraphCsvReader(Path path1,Path path2, ExecutionEnvironment context)
+	{
+		this.path1 = path1;
+		this.path2 = path2;
+		this.VertexReader = new CsvReader(path1,context);
+		this.EdgeReader = new CsvReader(path2,context);
+		this.mapper=null;
+		this.executionContext=context;
+	}
+
+	public GraphCsvReader(Path path2, ExecutionEnvironment context)
+	{
+		this.path1=null;
+		this.path2 = path2;
+		this.EdgeReader = new CsvReader(path2,context);
+		this.VertexReader = null;
+		this.mapper = null;
+		this.executionContext=context;
+	}
+
+	public GraphCsvReader(Path path2,final MapFunction mapper, ExecutionEnvironment context)
+	{
+		this.path1=null;
+		this.path2 = path2;
+		this.EdgeReader = new CsvReader(path2,context);
+		this.VertexReader = null;
+		this.mapper = mapper;
+		this.executionContext=context;
+	}
+
+	public GraphCsvReader (String path2,ExecutionEnvironment context)
+	{
+		this(new Path(Preconditions.checkNotNull(path2, "The file path may not be null.")), context);
+
+	}
+
+	public GraphCsvReader(String path1, String path2, ExecutionEnvironment context)
+	{
+		this(new Path(Preconditions.checkNotNull(path1, "The file path may not be null.")),new Path(Preconditions.checkNotNull(path2, "The file path may not be null.")), context);
+	}
+
+
+	public GraphCsvReader (String path2, final MapFunction mapper, ExecutionEnvironment context)
+	{
+
+			this(new Path(Preconditions.checkNotNull(path2, "The file path may not be null.")),mapper, context);
+
+
+	}
+
+	public CsvReader getEdgeReader()
+	{
+		return this.EdgeReader;
+	}
+
+	public CsvReader getVertexReader()
+	{
+		return this.VertexReader;
+	}
+	//--------------------------------------------------------------------------------------------------------------------
+
+	/**
+	 * Specifies the types for the Graph fields and returns a Graph with those field types
+	 *
+	 * This method is overloaded for the case in which Vertices don't have a value
+	 *
+	 * @param type0 The type of CSV field 0 and 1 for the CSV reader used for reading Edge data, the type of CSV field 0 for the CSV reader used for reading Vertex data  and the type of Vetex ID in the returned Graph.
+	 * @param  type1 The type of CSV field 1 for the CSV reader used for reading Vertex data and the type of Vertex Value in the returned Graph.
+	 * @param  type2 The type of CSV field 2 for the CSV reader used for reading Edge data and the type of Edge Value in the returned Graph.
+	 * @return The {@link org.apache.flink.graph.Graph} with Edges and Vertices extracted from the parsed CSV data.
+	 */
+	public <K,VV,EV> Graph<K,VV,EV> types(Class<K> type0, Class<VV> type1, Class<EV> type2)
+	{
+		DataSet<Tuple3<K,K,EV>> edges = this.EdgeReader.types(type0,type0,type2);
+		if(path1!=null)
+		{
+			DataSet<Tuple2<K,VV>> vertices = this.VertexReader.types(type0,type1);
+			return Graph.fromTupleDataSet(vertices,edges,executionContext);
+		}
+		else
+		{
+			return Graph.fromTupleDataSet(edges,mapper,executionContext);
```

Comment:
Could you please elaborate on which types? Should I pass types to the mapper function here too? The functions seemed to work without the type arguments for MapFunction so I thought it would be okay. I will work on this issue.